task/sast-snyk-check-oci-ta/0.3/sast-snyk-check-oci-ta.yaml

```diff
@@ -0,0 +1,204 @@
+      description: URL from repository to download known false positives files
+      type: string
+      default: ""
+    - name: PROJECT_NVR
+      description: Name-Version-Release (NVR) of the scanned project, used
+        to find path exclusions (it is optional)
+      type: string
+      default: ""
+    - name: RECORD_EXCLUDED
+      description: Write excluded records in file. Useful for auditing (defaults
+        to false).
+      type: string
+      default: "false"
+    - name: SNYK_SECRET
+      description: Name of secret which contains Snyk token.
+      default: snyk-secret
+    - name: SOURCE_ARTIFACT
+      description: The Trusted Artifact URI pointing to the artifact with
+        the application source code.
+      type: string
+    - name: caTrustConfigMapKey
+      description: The name of the key in the ConfigMap that contains the
+        CA bundle data.
+      type: string
+      default: ca-bundle.crt
+    - name: caTrustConfigMapName
+      description: The name of the ConfigMap to read CA bundle data from.
+      type: string
+      default: trusted-ca
+    - name: image-url
+      description: Image URL.
+      type: string
+      default: ""
+  results:
+    - name: TEST_OUTPUT
+      description: Tekton task test output.
+  volumes:
+    - name: snyk-secret
+      secret:
+        optional: true
+        secretName: $(params.SNYK_SECRET)
+    - name: trusted-ca
+      configMap:
+        items:
+          - key: $(params.caTrustConfigMapKey)
+            path: ca-bundle.crt
+        name: $(params.caTrustConfigMapName)
+        optional: true
+    - name: workdir
+      emptyDir: {}
+  stepTemplate:
+    volumeMounts:
+      - mountPath: /var/workdir
+        name: workdir
+  steps:
+    - name: use-trusted-artifact
+      image: quay.io/redhat-appstudio/build-trusted-artifacts:latest@sha256:e0e457b6af10e44ff6b90208a9e69adc863a865e1c062c4cb84bf3846037d74d
+      args:
+        - use
+        - $(params.SOURCE_ARTIFACT)=/var/workdir/source
+        - $(params.CACHI2_ARTIFACT)=/var/workdir/cachi2
+    - name: sast-snyk-check
+      image: quay.io/redhat-appstudio/konflux-test:v1.4.7@sha256:cf6808a3bd605630a5d9f20595ff7c43f8645c00381219d32f5a11e88fe37072
+      workingDir: /var/workdir/source
+      volumeMounts:
+        - mountPath: /etc/secrets
+          name: snyk-secret
+          readOnly: true
+        - mountPath: /mnt/trusted-ca
+          name: trusted-ca
+          readOnly: true
+      env:
+        - name: SNYK_SECRET
+          value: $(params.SNYK_SECRET)
+        - name: ARGS
+          value: $(params.ARGS)
+        - name: IMP_FINDINGS_ONLY
+          value: $(params.IMP_FINDINGS_ONLY)
+        - name: KFP_GIT_URL
+          value: $(params.KFP_GIT_URL)
+        - name: PROJECT_NVR
+          value: $(params.PROJECT_NVR)
+        - name: RECORD_EXCLUDED
+          value: $(params.RECORD_EXCLUDED)
+      script: "#!/usr/bin/env bash\n\nset -euo pipefail\n# shellcheck source=/dev/null\n.
```

Comment:
@jperezdealgaba I do not think the script should be obfuscated?  It used to be human-readable in the 0.2 version: https://github.com/konflux-ci/build-definitions/blob/31180b4b4c06e971dcaaac34ec2cfa62355469d9/task/sast-snyk-check-oci-ta/0.2/sast-snyk-check-oci-ta.yaml#L78

Reply:
Sure. Not sure how that happeneed. It was fixed

Reply:
Unresolving this. Just found out why this happened. In the "Trusted Artifacts variants" CI task, the error shows this message: 

```
Error: File is out of date, run `hack/generate-ta-tasks.sh` and include the updated file with your changes
Notice: Download the attached `ta.patch` file and run `git apply ta.patch`
```

Once applying the ta.patch, the obfuscated string script is generated. You can see it here: https://github.com/konflux-ci/build-definitions/actions/runs/11362215386/job/31604655663?pr=1359

Which file should I leave?

Reply:
@lcarva Could you please advise how to create the `oci-ta` version properly?

Reply:
@lcarva Once the ` ta.patch` file is applied, the script section of the `Task` gets obfuscated. That changed is even shown in the pipeline log: https://github.com/konflux-ci/build-definitions/actions/runs/11362708701/job/31605247099?pr=1359.

The previous version of the snyk-oci-ta task are not obfuscated, that's the reason we are asking, as it seems inconsistent with previous versions of the task

Reply:
The generated file might be ok. It's just using an inline string with `\n` instead of the nicer-looking `|`.

Reply:
The diff, at least, looks reasonable:

```diff
🐚 diff <(< task/sast-snyk-check/0.3/sast-snyk-check.yaml yq '.spec.steps[0].script') <(< task/sast-snyk-check-oci-ta/0.3/sast-snyk-check-oci-ta.yaml yq '.spec.steps[1].script')
32c32
< SOURCE_CODE_DIR=$(workspaces.workspace.path)
---
> SOURCE_CODE_DIR=/var/workdir
40c40
< snyk code test $ARGS --severity-threshold="$SEVERITY_THRESHOLD" "$SOURCE_CODE_DIR" --max-depth=1 --sarif-file-output=sast_snyk_check_out.json 1>&2>> stdout.txt
---
> snyk code test $ARGS --severity-threshold="$SEVERITY_THRESHOLD" "$SOURCE_CODE_DIR" --max-depth=1 --sarif-file-output=sast_snyk_check_out.json 1>&2 >>stdout.txt
48,50c48,50
< (cd "$SOURCE_CODE_DIR" && csgrep --mode=json --embed-context=3 "$(workspaces.workspace.path)"/hacbs/"$(context.task.name)"/sast_snyk_check_out.json) \
<   | csgrep --mode=json --strip-path-prefix="source/"  \
<   > sast_snyk_check_out_all_findings.json
---
> (cd "$SOURCE_CODE_DIR" && csgrep --mode=json --embed-context=3 "/var/workdir"/hacbs/"$(context.task.name)"/sast_snyk_check_out.json) |
>     csgrep --mode=json --strip-path-prefix="source/" \
>     >sast_snyk_check_out_all_findings.json
53c53,56
< (set -x; csgrep --mode=evtstat sast_snyk_check_out_all_findings.json)
---
> ( 
>   set -x
>          csgrep --mode=evtstat sast_snyk_check_out_all_findings.json
> )
77c80
<   "${CMD[@]}" sast_snyk_check_out_all_findings.json > filtered_sast_snyk_check_out.json
---
>   "${CMD[@]}" sast_snyk_check_out_all_findings.json >filtered_sast_snyk_check_out.json
87c90,93
<   (set -x; csgrep --mode=evtstat filtered_sast_snyk_check_out.json)
---
>   ( 
>     set -x
>            csgrep --mode=evtstat filtered_sast_snyk_check_out.json
>   )
90c96
< csgrep --mode=sarif filtered_sast_snyk_check_out.json > sast_snyk_check_out.sarif
---
> csgrep --mode=sarif filtered_sast_snyk_check_out.json >sast_snyk_check_out.sarif

```

Reply:
> But why don't previous versions of the tasks have the same format

I don't know. And it also doesn't happen to the script on the next step. I wonder if it's because the lines in the original script are too long?

Reply:
These lines are causing it:

```
(set -x; csgrep --mode=evtstat sast_snyk_check_out_all_findings.json)
```

```
(set -x; csgrep --mode=evtstat filtered_sast_snyk_check_out.json)
```

Reply:
Changing them to `set -x && csgrep` seems to generate a better script.

Reply:
Thank you very much for that! I will try it with those changes!